### ISSUE TITLE
bug: `datagen` connector should not be used with `ROW FORMAT JSON`

### ISSUE THREAD

**Issue by fuyufjh** [risingwavelabs/risingwave#3307]
**Describe the bug**

```
dev=> create source ss (id int) with ('connector' = 'datagen');
ERROR:  sql parser error: Expected ROW, found: ;
```

But this works:

```
dev=> create source ss (id int) with ('connector' = 'datagen') ROW FORMAT JSON;
CREATE_SOURCE
```

**To Reproduce**

See above.

**Expected behavior**

`datagen` connector should not be used with `ROW FORMAT JSON`

**Additional context**

**Comment by fuyufjh** [comment]
@shanicky PTAL 🥰

**Comment by shanicky** [comment]
The connector crate only generates raw bytes messages that need to be converted into chunks by the source parser, so this is a problem for this internal connector.
Let's figure it out.

cc @tabVersion

**Comment by shanicky** [comment]
fixed in #7612
